src/test/java/codeu/controller/LoginServletTest.java

```diff
@@ -80,26 +70,35 @@ public void testDoPost_NewUser() throws IOException, ServletException {
 
     Mockito.verify(mockUserStore).addUser(userArgumentCaptor.capture());
     Assert.assertEquals(userArgumentCaptor.getValue().getName(), "test username");
-
+    Assert.assertEquals(userArgumentCaptor.getValue().getPassword(), "test password");
+    
     Mockito.verify(mockSession).setAttribute("user", "test username");
+    Mockito.verify(mockSession).setAttribute("password", "test password");
+    
     Mockito.verify(mockResponse).sendRedirect("/conversations");
   }
 
   @Test
   public void testDoPost_ExistingUser() throws IOException, ServletException {
     Mockito.when(mockRequest.getParameter("username")).thenReturn("test username");
-
+    Mockito.when(mockRequest.getParameter("password")).thenReturn("test password");
+    
     UserStore mockUserStore = Mockito.mock(UserStore.class);
     Mockito.when(mockUserStore.isUserRegistered("test username")).thenReturn(true);
     loginServlet.setUserStore(mockUserStore);
-
+    
     HttpSession mockSession = Mockito.mock(HttpSession.class);
     Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
-
+    
+    User mockUser = Mockito.mock(User.class);
+    Mockito.when(mockUser.getName()).thenReturn("test username");
+    Mockito.when(mockUser.getPassword()).thenReturn("test password");
+    Mockito.verify(mockUserStore).addUser(mockUser);
```

Comment:
A new user shouldn't be created when an existing user logs in.

Reply:
![image](https://user-images.githubusercontent.com/10622320/37635533-50a27230-2bb9-11e8-8d36-ed74a7ab35a9.png)
I tried to only use mockUserStore to set the username and password, but that would give me null pointer exception because when I call mockUserStore.getUser("username"), it is null.  Also, I am not sure assertEquals or thenReturn, do they yield the same result? Do I need to do another mockito.when this is called an return a user class? Or does that still count as creating a new User? 
As for testing the wrong password, would it work if I just have mockito return a wrong password as follows?
![image](https://user-images.githubusercontent.com/10622320/37635612-d54298f8-2bb9-11e8-8ae5-7dfb98e6daff.png)